### ISSUE TITLE
terraform init for 4.88.0 gets 404 on macOS

### ISSUE THREAD

**Issue by infa-ddeore** [oracle/terraform-provider-oci#1650]
### Affected Resource(s)
`v.4.88.0` provider

### Panic Output
```
╷
│ Error: Failed to install provider
│ 
│ Error while installing oracle/oci v4.88.0: could not query provider registry for registry.terraform.io/oracle/oci: failed to retrieve authentication checksums for
│ provider: 404 Not Found returned from github.com
```

### Expected Behavior
it should download the provider

### Actual Behavior
getting 404

### Steps to Reproduce
1. create `main.tf`
```
terraform {
  required_providers {
    oci = {
      source  = "oracle/oci"
      version = "4.88.0"
    }
  }
}
```
2. `terraform init`

### Important Factoids
issue is with the links in `https://registry.terraform.io/v1/providers/oracle/oci/4.88.0/download/darwin/amd64`

actual files have **"v"** in the version `v4.88.0` but links in the payload dont have it:
```
$ curl -s https://registry.terraform.io/v1/providers/oracle/oci/4.88.0/download/darwin/amd64 | jq '. | .download_url, .shasums_url'
"https://github.com/oracle/terraform-provider-oci/releases/download/4.88.0/terraform-provider-oci_4.88.0_darwin_amd64.zip"
"https://github.com/oracle/terraform-provider-oci/releases/download/4.88.0/terraform-provider-oci_4.88.0_SHA256SUMS"
```

whereas `4.87.0` version has correct links so it doesnt have any issue
```
$ curl -s https://registry.terraform.io/v1/providers/oracle/oci/4.87.0/download/darwin/amd64 | jq '. | .download_url, .shasums_url'
"https://github.com/oracle/terraform-provider-oci/releases/download/v4.87.0/terraform-provider-oci_4.87.0_darwin_amd64.zip"
"https://github.com/oracle/terraform-provider-oci/releases/download/v4.87.0/terraform-provider-oci_4.87.0_SHA256SUMS"
```

**Comment by cfstras** [comment]
I've seen this as well, both in GitHub actions (linux/amd64) and MacOS (darwin/arm64 & darwin/amd64), when specifying a version like `">= 4.0.0"`. Pinning the version to an earlier version works.

**Comment by junior** [comment]
For reference, if you use `hashicorp/oci` instead, you can use the version 4.88.0 without problems, just shows the expected warning to move to the `oracle/oci`

**Comment by junior** [comment]
@infa-ddeore using you factoids, appear that something changed. If you use the same method with `hashicorp/oci`, no version uses the "v" and both 4.87 and 4.88 works.

```
ᛡ curl -s https://registry.terraform.io/v1/providers/hashicorp/oci/4.88.0/download/darwin/amd64 | jq '. | .download_url, .shasums_url'
"https://releases.hashicorp.com/terraform-provider-oci/4.88.0/terraform-provider-oci_4.88.0_darwin_amd64.zip"
"https://releases.hashicorp.com/terraform-provider-oci/4.88.0/terraform-provider-oci_4.88.0_SHA256SUMS"
```

```
ᛡ curl -s https://registry.terraform.io/v1/providers/hashicorp/oci/4.87.0/download/darwin/amd64 | jq '. | .download_url, .shasums_url'
"https://releases.hashicorp.com/terraform-provider-oci/4.87.0/terraform-provider-oci_4.87.0_darwin_amd64.zip"
"https://releases.hashicorp.com/terraform-provider-oci/4.87.0/terraform-provider-oci_4.87.0_SHA256SUMS"
```

**Comment by eamonnlambda** [comment]
+1. Encountering this on Terraform Cloud, Linux, and Mac systems. Almost seems like someone deleted v4.88.0. Terraform Registry also shows a similar error - I'll see if I can find a link that demonstrates it concisely

**Comment by mcint** [comment]
I pinned the previous version, 4.87, but it expects 4.88. I might be able to fix or workaround this by updating my project's lockfile.
![Screen Shot 2022-08-11 at 1 09 43 PM](https://user-images.githubusercontent.com/601708/184234222-6af0cc36-d86d-4b6f-89e9-f3b4429e0c13.png)

**Comment by mcint** [comment]
Fixed by https://github.com/oracle/terraform-provider-oci/pull/1651 ...once the release gets published 😁

**Comment by eamonnlambda** [comment]
Confirmed #1651 fixes this.

**Comment by mcint** [comment]
Fixed by `4.88.1`, now released.

**Comment by junior** [comment]
Tested and is working as expect.

**Comment by tf-oci-pub** [comment]
Fixed with version 4.88.1

**Comment by luissimoesneom** [comment]
My goal was to understand how the OCI Resource Manager performs the authentication if we create a new project...
Can anyone assist?
